softlayer/vm/pool/vm_pool_manager.go

```diff
@@ -0,0 +1,311 @@
+	}
+	return nil
+}
+
+func (vmInfoDB *VMInfoDB) QueryVMInfobyAgentID(retryTimeout time.Duration, retryInterval time.Duration) error {
+
+	execStmtRetryable := boshretry.NewRetryable(
+		func() (bool, error) {
+			tx, err := vmInfoDB.db.Begin()
+			if err != nil {
+				sqliteErr := err.(sqlite3.Error)
+				if sqliteErr.Code == sqlite3.ErrBusy || sqliteErr.Code == sqlite3.ErrLocked {
+					return true, bosherr.WrapError(sqliteErr, "retrying...")
+				} else {
+					return false, bosherr.WrapError(sqliteErr, "Failed to begin DB transcation")
+				}
+			}
+
+			var prepareStmt string
+			if vmInfoDB.VmProperties.InUse == "t" {
+				prepareStmt = "select id, image_id, agent_id from vms where in_use='t' and agent_id=?"
+			} else if vmInfoDB.VmProperties.InUse == "f" {
+				prepareStmt = "select id, image_id, agent_id from vms where in_use='f' and agent_id=?"
+			} else {
+				prepareStmt = "select id, image_id, agent_id from vms where agent_id==?"
+			}
+
+			sqlStmt, err := tx.Prepare(prepareStmt)
+			defer sqlStmt.Close()
+			if err != nil {
+				sqliteErr := err.(sqlite3.Error)
+				if sqliteErr.Code == sqlite3.ErrBusy || sqliteErr.Code == sqlite3.ErrLocked {
+					return true, bosherr.WrapError(sqliteErr, "retrying...")
+				} else {
+					return false, bosherr.WrapError(sqliteErr, "Failed to prepare sql statement")
+				}
+			}
+
+			err = sqlStmt.QueryRow(vmInfoDB.VmProperties.AgentId).Scan(&vmInfoDB.VmProperties.Id, &vmInfoDB.VmProperties.ImageId, &vmInfoDB.VmProperties.AgentId)
+			if err != nil && !strings.Contains(err.Error(), "no rows") {
+				sqliteErr := err.(sqlite3.Error)
+				if sqliteErr.Code == sqlite3.ErrBusy || sqliteErr.Code == sqlite3.ErrLocked {
+					return true, bosherr.WrapError(sqliteErr, "retrying...")
+				} else {
+					return false, bosherr.WrapError(sqliteErr, "Failed to query VM info from vms table")
+				}
+
+			}
+			tx.Commit()
+			return false, nil
+		})
+
+	timeService := clock.NewClock()
+	timeoutRetryStrategy := boshretry.NewTimeoutRetryStrategy(retryTimeout, retryInterval, execStmtRetryable, timeService, vmInfoDB.logger)
+	err := timeoutRetryStrategy.Try()
+	if err != nil {
+		return bosherr.WrapError(err, fmt.Sprintf("Failed to run QueryVMInfobyAgentID"))
+	} else {
+		return nil
+	}
+
+}
+
+func (vmInfoDB *VMInfoDB) QueryVMInfobyID(retryTimeout time.Duration, retryInterval time.Duration) error {
+
+	execStmtRetryable := boshretry.NewRetryable(
+		func() (bool, error) {
+			tx, err := vmInfoDB.db.Begin()
+			if err != nil {
+				sqliteErr := err.(sqlite3.Error)
+				if sqliteErr.Code == sqlite3.ErrBusy || sqliteErr.Code == sqlite3.ErrLocked {
+					return true, bosherr.WrapError(sqliteErr, "retrying...")
+				} else {
+					return false, bosherr.WrapError(sqliteErr, "Failed to begin DB transcation")
+				}
+			}
+
+			var prepareStmt string
+			if vmInfoDB.VmProperties.InUse == "t" {
+				prepareStmt = "select id, in_use, image_id, agent_id from vms where id=? and in_use='t'"
```

Comment:
In SQL I typically use all upper case for the SQL statements, e.g., `SELECT id, in_use, ...`